service/src/test/kotlin/app/cash/backfila/service/BackfillRunnerTest.kt

```diff
@@ -604,30 +552,23 @@ class BackfillRunnerTest {
     fakeBackfilaClientServiceClient.dontBlockGetNextBatch()
     val runner = startBackfill(numThreads = 1, extraSleepMs = 1000L)
 
-    runBlocking {
-      launch { runner.run() }
-      try {
-        fakeBackfilaClientServiceClient.runBatchRequests.receive()
-        fakeBackfilaClientServiceClient.runBatchResponses.send(
-          Result.success(RunBatchResponse.Builder().build())
-        )
+    runBlockingTestCancellable {
+      launch { runner.start(this) }
 
-        delay(500)
-        // Nothing sent yet - the backoff is 1000ms
-        assertThat(fakeBackfilaClientServiceClient.runBatchRequests.poll()).isNull()
-        // Give a bit more buffer to send next request due to scheduling delays.
-        assertThat(
-          withTimeoutOrNull(2000) {
-            fakeBackfilaClientServiceClient.runBatchRequests.receive()
-          }
-        ).isNotNull()
-        fakeBackfilaClientServiceClient.runBatchResponses.send(
-          Result.success(RunBatchResponse.Builder().build())
-        )
-        delay(2000)
-      } finally {
-        runner.stop()
-      }
+      fakeBackfilaClientServiceClient.runBatchRequests.receive()
+      fakeBackfilaClientServiceClient.runBatchResponses.send(
+        Result.success(RunBatchResponse.Builder().build())
+      )
+
+      delay(500)
+      // Nothing sent yet - the backoff is 1000ms
+      assertThat(fakeBackfilaClientServiceClient.runBatchRequests.poll()).isNull()
+      delay(500)
+      assertThat(fakeBackfilaClientServiceClient.runBatchRequests.poll()).isNotNull()
+      fakeBackfilaClientServiceClient.runBatchResponses.send(
+        Result.success(RunBatchResponse.Builder().build())
+      )
+      runner.stop()
```

Comment:
stop doesn't have to be in the finally?

Reply:
see the thing I made `runBlockingTestCancellable` it will cancel all children if any assert fails, so we no longer have to stop() in finally